cfme/automate/service_dialogs.py

```diff
@@ -95,148 +120,159 @@ def __init__(self, label=None, description=None,
 
     def update(self, updates):
-        navigate_to(self, 'Edit')
-        fill(label_form, {'name_text': updates.get('name', None),
-                          'description_text': updates.get('description', None)})
-        sel.click(form_buttons.save)
-        flash.assert_no_errors()
+        view = navigate_to(self, 'Edit')
+        changed = view.fill(updates)
+        if changed:
+            view.save_button.click()
+        else:
+            view.cancel_button.click()
+        view = self.create_view(DetailsDialogView, override=updates)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        if changed:
+            view.flash.assert_message(
+                'Dialog "{}" was saved'.format(updates.get('name', self.label)))
+        else:
+            view.flash.assert_message(
+                'Edit of Dialog "{}" was cancelled by the user'.format(self.label))
 
     def update_element(self, second_element, element_data):
-        navigate_to(self, 'Edit')
-        if self.appliance.version > "5.5":
-            tree = accordion.tree("Dialog")
-        else:
-            tree = Tree("dialog_edit_treebox")
-        tree.click_path(self.label, self.tab_label, self.box_label)
-        self.add_element(second_element)
+        view = navigate_to(self, 'Edit')
+        view.element_tree.click_path(self.label, self.tab_label, self.box_label)
+        self.add_element(view, second_element)
         list_ele = []
         list_ele.append(element_data.get("ele_label"))
         list_ele.append(second_element.get("ele_label"))
-        tree.click_path(self.label, self.tab_label, self.box_label)
+        view.element_tree.click_path(self.label, self.tab_label, self.box_label)
         ele_1 = self.element(list_ele[0])
         ele_2 = self.element(list_ele[1])
         sel.drag_and_drop(ele_1, ele_2)
-        sel.click(form_buttons.save)
-        flash.assert_no_errors()
-
-    def delete(self, cancel=False):
-        navigate_to(self, 'Details')
-        cfg_btn(version.pick({version.LOWEST: "Remove from the VMDB",
-                              '5.7': 'Remove Dialog'}),
-                invokes_alert=True)
-        sel.handle_alert(cancel)
-
-    def element_type(self, each_element):
+        view.save_button.click()
+        view = self.create_view(DetailsDialogView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+
+    def delete(self):
+        view = navigate_to(self, "Details")
+        view.configuration.item_select(
+            version.pick({
+                version.LOWEST: 'Remove from the VMDB',
+                '5.7': 'Remove Dialog'}),
+            handle_alert=True)
+        view = self.create_view(DialogsView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_success_message(
+            'Dialog "{}": Delete successful'.format(self.label))
+
+    def element_type(self, view, each_element):
```

Comment:
would it make sense to create class and view for each control and operate by classes ?
You will be able to define default values inside appropriate classes, etc.